core/lib/zksync_core/src/lib.rs

```diff
@@ -122,6 +124,26 @@ pub async fn genesis_init(
     )
     .context("Failed to restore operator address from private key")?;
 
+    // We don't need to worry about backward-compatibility with the old verifier here,
+    // since this is only run during genesis.
+    let eth_client = QueryClient::new(eth_client_url)?;
```

Comment:
I'm introducing a flag in https://github.com/matter-labs/zksync-era/pull/158 - so you could depend on it - this way we'd still be backwards compatible.